### PR TITLE
a8n: Align ChangsetPlan/ExternalChangset interfaces

### LINKED ISSUE #6752: a8n: Align API for previews and external changesets
I know I signed this off earlier, but it turns out there are quite a few places where `ChangesetPlan` and `ExternalChangeset` are used in the same context, but they currently are not very similar in their structure. 

For example, in the WebApp we have to always check the `__typename` property before we can use the diff:

```ts
const diffStat = campaign
        ? campaign.__typename === 'CampaignPlan'
            ? campaign.diff.fileDiffs.diffStat
            : campaign.fileDiffs.diffStat
        : null
```

current schema
```graphql
type ChangesetPlan {
    repository: Repository!
    fileDiffs(first: Int): PreviewFileDiffConnection!
}

type PreviewFileDiffConnection {
    nodes: [PreviewFileDiff!]!
    totalCount: Int
    pageInfo: PageInfo!
    diffStat: DiffStat!
    rawDiff: String!
}
```

```graphql
type ExternalChangeset {
    repository: Repository!
    diff: RepositoryComparison
	... others
}

type RepositoryComparison {
    baseRepository: Repository!
    headRepository: Repository!
    range: GitRevisionRange!
    commits(first: Int): GitCommitConnection!
    fileDiffs(first: Int): FileDiffConnection!
}

type FileDiffConnection {
    nodes: [FileDiff!]!
    totalCount: Int
    pageInfo: PageInfo!
    diffStat: DiffStat!
    rawDiff: String!
}
```

If we would reintroduce the `RepositoryComparison` on the preview level, the structure would be more aligned and it would help a lot creating more broadly usable components for both previews and actual campaigns/changesets :)

So I'm suggesting here that we reintroduce 
```graphql
type PreviewRepositoryComparison {
    baseRepository: Repository!
    fileDiffs(first: Int): PreviewFileDiffConnection!
}
```

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes #6752 by reintroducing the PreviewRepositoryComparison (previously called: PreviewRepositoryDiff) interface on ChangsetPlan.

The campaignJobResolver implements this interface without having the need to define another resolver.

--- 

@eseliger: Let me know if that's what you had in mind.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -624,6 +624,9 @@ type ChangesetPlan {
 
     # The preview of the file diffs for each file in the diff.
     fileDiffs(first: Int): PreviewFileDiffConnection!
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/schema.graphql]: wouldn't that be removed then?

Comment (mrnugget) [reply cmd/frontend/graphqlbackend/schema.graphql]: Depends on what you want 😃 If I should remove it, I'll remove it.

Comment (eseliger) [reply cmd/frontend/graphqlbackend/schema.graphql]: Removal is fine 😄 keeps the api simple and adds no duplicates. I have to adjust some frontend stuff to umbrellas the build. If you can adjust it I will commit to this branch from the train later today

Comment (mrnugget) [reply cmd/frontend/graphqlbackend/schema.graphql]: Adjusted & pushed. Feel free to add to this branch.

**Pull request merged**